File: api/HomeBudget.API/HomeBudget.Application/DTOs/CategoriaDtos/CategoriaDto.cs
using HomeBudget.Domain.Enums;

namespace HomeBudget.Application.DTOs.CategoriaDtos
{
    public class CategoriaDto
    {
        public Guid Id { get; set; }
        public List<Guid> IdTransacoes { get; set; } = [];

        public required string Nome { get; set; }
        public string? Descricao { get; set; }

        public TipoCategoria Finalidade { get; set; }


    }
}

File: web/src/interfaces/CategoriaDtos/CategoriaDto.ts
import { TipoCategoria, TipoCategoriaLabel } from "../../enums/TipoCategoriaEnum";
import { empty_Guid } from "../../utils/guid";

export class CategoriaDto {
  id: string;
  nome: string = "";
  descricao: string = "";
  finalidade: string = TipoCategoriaLabel[TipoCategoria.Despesa];
  dataCriacaoModificacao: string = new Date().toISOString();

  constructor(
    id: string = empty_Guid,
    nome: string = "",
    descricao: string = "",
    finalidade: string = TipoCategoriaLabel[TipoCategoria.Despesa],
    dataCriacaoModificacao: string = new Date().toISOString(),
  ) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.finalidade = finalidade;
    this.dataCriacaoModificacao = dataCriacaoModificacao;
  }
}
